profitbricks/src/main/java/org/jclouds/profitbricks/domain/LoadBalancer.java

```diff
@@ -0,0 +1,398 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jclouds.profitbricks.domain;
+
+import com.google.auto.value.AutoValue;
+import com.google.common.collect.ImmutableList;
+import org.jclouds.javax.annotation.Nullable;
+
+import java.util.Date;
+import java.util.List;
+
+@AutoValue
+public abstract class LoadBalancer {
+
+   public enum LoadbalancerAlgorithm {
+
+      ROUND_ROBIN, UNRECOGNIZED;
+
+      public static LoadbalancerAlgorithm
+              fromValue(String value) {
+         try {
+            return valueOf(value);
+         } catch (IllegalArgumentException e) {
+            return UNRECOGNIZED;
+         }
+      }
+   }
+
+   @Nullable
+   public abstract String id();
+
+   @Nullable
+   public abstract String name();
+
+   @Nullable
+   public abstract LoadbalancerAlgorithm loadBalancerAlgorithm();
+
+   @Nullable
+   public abstract String dataCenterId();
+
+   @Nullable
+   public abstract String dataCenterVersion();
+
+   @Nullable
+   public abstract boolean internetAccess();
+
+   @Nullable
+   public abstract String ip();
+
+   @Nullable
+   public abstract String lanId();
+
+   @Nullable
+   public abstract ProvisioningState state();
+
+   @Nullable
+   public abstract Date creationTime();
+
+   @Nullable
+   public abstract Date lastModificationTime();
+
+   @Nullable
+   public abstract List<Server> balancedServers();
+
+   @Nullable
+   public abstract List<Firewall> firewalls();
```

Comment:
Are **all** fields of the LoadBalancer really nullable? Is it possible that there is a load balancer without name, without id, algorithm, etc? If not, configure `@Nullable` annoations properly.